### ISSUE TITLE
Error in sentry_sync.c file

### ISSUE THREAD

**Issue by rafagan** [getsentry/sentry-native#174]
I have the following error when building:

PS C:\Users\rafagan\Desktop\sentry-native> cmake --build build --parallel
Microsoft (R) Build Engine version 15.5.180.51428 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  zlib.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\zlib.lib
  getopt.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\getopt.lib
  mini_chromium.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\mini_chromium.lib
  crashpad_compat.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\crashpad_compat.lib
  crashpad_tools.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\crashpad_tools.lib
  crashpad_util.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\crashpad_util.lib
  crashpad_client.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\crashpad_client.lib
  crashpad_snapshot.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\crashpad_snapshot.lib
  crashpad_minidump.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\crashpad_minidump.lib
  crashpad_handler.vcxproj -> C:\Users\rafagan\Desktop\sentry-native\build\crashpad_handler.exe
  sentry_sync.c
C:\Users\rafagan\Desktop\sentry-native\src\sentry_sync.c(105): error C2440: 'function': cannot convert from 'int (__cde
cl *)(void *)' to 'LPTHREAD_START_ROUTINE' [C:\Users\rafagan\Desktop\sentry-native\build\sentry.vcxproj]
C:\Users\rafagan\Desktop\sentry-native\src\sentry_sync.c(105): warning C4024: 'CreateThread': different types for forma
l and actual parameter 3 [C:\Users\rafagan\Desktop\sentry-native\build\sentry.vcxproj]

How can I solve that?

**Comment by rafagan** [comment]
The error is occuring at the following lines (I commented the piece of code to build successfully):

```
void
sentry__bgworker_start(sentry_bgworker_t *bgw)
{
    SENTRY_TRACE("starting background worker thread");
    sentry__mutex_lock(&bgw->task_lock);
    bgw->running = true;
    //if (sentry__thread_spawn(&bgw->thread_id, &worker_thread, bgw) != 0) {
    //    bgw->running = false;
    //}
    sentry__mutex_unlock(&bgw->task_lock);
}
```

**Comment by Swatinem** [comment]
@rafagan I think PR #168 contains some fixes related to calling conventions which look like they are related.

As you created quite a bunch of reports, can you please write the exact OS/toolchain/compiler versions that you are using? `cmake` and `cmake build` output different versions.

**Comment by Swatinem** [comment]
```
-- Building for: Visual Studio 16 2019
-- The C compiler identification is MSVC 19.22.27905.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.22.27905/bin/Hostx64/x64/cl.exe
[…]
Microsoft (R)-Build-Engine, Version 16.2.37902+b5aaefc9f für .NET Framework
```

I think I have the latest versions of everything. Not quite sure what versions are running on Azure, but we use the `windows-latest` image there as well, so probably quite recent versions as well.

If you have some older/outdated versions, I can take a look at reproducing this locally for me.

**Comment by rafagan** [comment]
Try with Microsoft (R) Build Engine version 15.5.180.51428 (MSVC++ 2017) x64 and cmake 3.17.0-rc2.

**Comment by Swatinem** [comment]
ah yes, I have pulled the fix from #168 into #176 , which at least fixes this build failure on VS2017.
We will extend our testsuite to cover more compiler permutations, and will clean up all this along the way

**Comment by rafagan** [comment]
I'll try again today with MSVC++2019 and then return here to post status
